src/helper-modules/companies-house/wrap-companies-house-http-error-callback.ts

```diff
@@ -9,21 +9,25 @@ type CompaniesHouseHttpErrorCallback = (error: Error) => ObservableInput<never>;
 export const createWrapCompaniesHouseHttpGetErrorCallback =
   ({ messageForUnknownError, knownErrors }: { messageForUnknownError: string; knownErrors: KnownErrors }): CompaniesHouseHttpErrorCallback =>
   (error: Error) => {
-    if (error instanceof AxiosError && error.response && typeof error.response.data === 'object') {
-      const errorResponseData = error.response.data;
-      let errorMessage: string;
+    if (error instanceof AxiosError) {
+      let message: string;
+      const status: number = error?.response?.status;
 
-      if (typeof errorResponseData.error === 'string') {
-        errorMessage = errorResponseData.error;
-      } else if (errorResponseData.errors && errorResponseData.errors[0] && typeof errorResponseData.errors[0].error === 'string') {
-        errorMessage = errorResponseData.errors[0].error;
+      if (typeof error?.response?.data?.error === 'string') {
+        message = error.response.data.error;
+      } else if (error?.response?.data?.errors[0] && typeof error?.response?.data?.errors[0]?.error === 'string') {
+        message = error.response.data.errors[0].error;
       }
 
-      if (errorMessage) {
-        const errorMessageInLowerCase = errorMessage.toLowerCase();
-
-        knownErrors.forEach(({ caseInsensitiveSubstringToFind, throwError }) => {
-          if (errorMessageInLowerCase.includes(caseInsensitiveSubstringToFind.toLowerCase())) {
+      /**
+       * CH API will throw empty response body with `404`
+       * upon sending a valid non-existent 8 digit CRN.
+       *
+       * Otherwise `404` with an expected response body.
+       */
+      if (message || status) {
```

Comment:
I think we should/could remove this check as it's coupling the inner workings of our error handling instances with this function. In an ideal world, this function does not care about the errors it's handling -- all it needs to know about the `knownErrors` is that it can call `checkHasError` on it.

The current implimentation does save some time technically, but I think at the cost of making the code more coupled where it doesn't need to be, and less extendable later... What do you think?

Reply:
This would also allow us to deduplicate the code on lines 16-19 of this file, as it would no longer be needed as we'd be decoupling this function from caring about it.

Reply:
Good shout, have refactored.